### PR TITLE
New abapLint Rules

### PR DESCRIPTION
**Pull request by lucasborin**

### CODE REVIEW

**Review by eugen-guenther-sap — DISMISSED** [review]

**Comment by eugen-guenther-sap** [comment]
whats with abaplint? 
45 issues just abaout the upper/lower case -.-

**Comment by larshp** [comment]
it needs additional configuration, or alternatively the findings to be fixed

https://rules.abaplint.org/keyword_case/

I also opened https://github.com/abaplint/abaplint/issues/1671 yesterday

**Review by eugen-guenther-sap — APPROVED** [review]

**Comment by larshp** [comment]
Hi,
I've looked through the 45 findings, and personally, I think they should be fixed.

But as noted in https://github.com/SAP/styleguides/blob/master/clean-abap/CleanABAP.md#use-your-pretty-printer-team-settings, its important that your team decides which style to go for. But looking at the current code its mainly upper case keywords.

**Comment by lucasborin** [comment]
Checks reviewed. Let's take a look at the reported issues now.

**Comment by larshp** [comment]
also remove following from abaplint.json, abapGit is required to install code-pal, but the code-pal implementation does not call any ZCL_ABAPGIT* code

```abap
    {
      "url": "https://github.com/abapGit/abapGit",
      "folder": "/deps2",
      "files": "/src/**/*.*"
    }
```

**Comment by lucasborin** [comment]
A few checks set to warning as they are reporting false positives. Please, check abapLint issues for further information.

File: src/checks/y_check_empty_catches.clas.abap [hunk 1]
```diff
@@ -31,7 +31,7 @@ CLASS Y_CHECK_EMPTY_CATCHES IMPLEMENTATION.
 
   METHOD get_next_token_from_index.
     LOOP AT ref_scan_manager->tokens ASSIGNING FIELD-SYMBOL(<token>)
-      FROM index WHERE type EQ 'I'.
+      FROM index WHERE type ='I'.
```
Comment (lucasborin) [inline src/checks/y_check_empty_catches.clas.abap:34]: Missing space.

**Review by lucasborin — COMMENTED** [review]
Review done.

**Comment by lucasborin** [comment]
@eugen-guenther-sap : Would you like to review the changed objects? 
@larshp : These 9 issues reported by abaplint are false positives (reported in abapLint issues)

**Review by eugen-guenther-sap — APPROVED** [review]

**Comment by larshp** [comment]
Try again, 2.73.8 published

Also note there are a few warnings in the configuration file,
image

**Comment by larshp** [comment]
🥳 🍾

**Pull request merged**